### PR TITLE
Remove updateModuleSettings from codebase

### PR DESCRIPTION
**Pull request by felixpalmer**
For https://github.com/visgl/deck.gl/issues/8997

To merge `fp64` module in luma needs to be [ported to UBO](https://github.com/visgl/luma.gl/pull/2262), for now a manual hack is included here

#### Change List
- Remove `updateModuleSettings` from `HeatmapLayer`
- Remove `updateModuleSettings` from `Layer`
- Pass `maskProps` via `ShaderInputs`
- Test fix

### CODE REVIEW

File: modules/core/src/lib/layer.ts [hunk 1]
```diff
@@ -1107,6 +1109,12 @@ export default abstract class Layer<PropsT extends {} = {}> extends Component<
           lightSources
         } = moduleParameters;
 
+        const maskProps = {
```
Comment (ibgreen) [inline modules/core/src/lib/layer.ts:1112]: Type the props?

File: modules/core/src/lib/layer.ts [hunk 2]
```diff
@@ -1129,22 +1137,25 @@ export default abstract class Layer<PropsT extends {} = {}> extends Component<
           terrainSkipRender
         };
 
+        const projectProps = {
+          viewport,
+          devicePixelRatio,
+          modelMatrix,
+          coordinateSystem,
+          coordinateOrigin
+        } as ProjectProps;
```
Comment (ibgreen) [inline modules/core/src/lib/layer.ts:1146]: If possible, we should avoid forced casts as they hide problems.
```suggestion
        } as const satisfies ProjectProps;
```

**Review by ibgreen — APPROVED** [review]
Very timely, as luma.gl 9.2 removes `updateModuleSettings`. If possible and haven't done so, search the code base and make sure that all references are removed (I think there were only 3 or so).

**Review by Pessimistress — APPROVED** [review]

File: modules/core/src/lib/layer.ts (hunk 1 above)
Comment (felixpalmer) [reply modules/core/src/lib/layer.ts:1113]: The type is in `deck.gl/extensions` and it is an open question whether this is the correct place to update the `ShaderInputs` as in many cases the referenced `ShaderModules` will not be present. Agree that we should type, will revisit when cleaning up, e.g. moving the `shaderInputs.setProps` into the extensions

**Pull request merged**